engine/skiplist.cpp

```diff
@@ -60,17 +60,28 @@ void SkiplistNode::SeekNode(const pmem::obj::string_view &key,
       if (next_next.GetTag()) {
         if (prev->CASNext(i, next, next_next.RawPointer())) {
           if (--next->valid_links == 0) {
-            if (to_delete == nullptr) {
-              to_delete.reset(new std::vector<SkiplistNode *>());
-            }
-            to_delete->push_back(next.RawPointer());
+            to_delete.push_back(next.RawPointer());
           }
         }
         // if prev is marked deleted before cas, cas will be failed, and prev
         // will be roll back in next round
         continue;
       }
-      int cmp = compare_string_view(key, next->UserKey());
+
+      DLRecord *next_pmem_record = next->record;
+      // pmem record maybe updated during seek, then the next record could be
+      // invalid, so we need to catch exceptions here
+      int cmp;
+      try {
```

Comment:
why need `catch`?

Reply:
because the next_pmem_record maybe modified, then its key maybe a invalid address

Reply:
because the next_pmem_record maybe modified, then its key maybe a invalid address

Reply:
Considering a situation： we can access the invalid address, but key is strange string,  this code can't catch this situation.

Reply:
This can not pass double check, the dram node will not point to it

Reply:
This can not pass double check, the dram node will not point to it

Reply:
I think we only need double check,  `catch` seems no need